### PR TITLE
Revises 8.x-8.x upgrade guide

### PR DESCRIPTION
**Pull request by natasha-moore-elastic**
Contributes to https://github.com/elastic/security-docs/issues/5302.

Revises the instructions for upgrading from an 8.x to an 8.x version. Mostly reuses the content from the [7.17 to 8.x upgrade guide](https://www.elastic.co/guide/en/security/master/upgrade-7.17-8x.html), but without the 7.17-specific information.

### CODE REVIEW

File: docs/upgrade/upgrade-security.asciidoc [hunk 1]
```diff
@@ -37,69 +37,104 @@ but upgrading from a pre-release to the Generally Available version is unsupport
...
+... If you use {ml}, ensure that it is up and running. 
+.. For {kib}: 
+... Ensure that you and your users can successfully log in to {kib} and access desired pages.
+... Check {kibana-ref}/discover.html[Discover] and verify that the index patterns you typically use are available.
+... Verify that your commonly used {kibana-ref}/dashboard.html[dashboards] are available and working properly.
+... If you use any Watcher-based {kib} scheduled {kibana-ref}/reporting-getting-started.html[reporting], ensure that it's working properly.
+
+. Upgrade your ingest components (such as {ls}, {fleet} and {agent}, {beats}, etc.). For details, refer to the {stack-ref}/upgrading-elastic-stack.html[Elastic Stack upgrade docs].
+
+. Validate that Ingest is operating correctly.
```
Comment (joepeeples) [inline docs/upgrade/upgrade-security.asciidoc:124]:
```suggestion
. Validate that ingest is operating correctly.
```

File: docs/upgrade/upgrade-security.asciidoc [hunk 2]
```diff
@@ -37,69 +37,104 @@ but upgrading from a pre-release to the Generally Available version is unsupport
...
+.. Open *Discover*, go through data views for each of your expected ingest data streams, and ensure that data is being ingested in the expected format and volume. 
+
+. Validate that {elastic-sec} is operating correctly.
+.. Re-enable your desired SIEM detection rules (rule management), and ensure that enabled rules are running without errors or warnings (rule monitoring).
```
Comment (joepeeples) [inline docs/upgrade/upgrade-security.asciidoc:128]: Seems like these might be references to specific UI tabs on the Rules page, so capitalize:
```suggestion
.. Re-enable your desired SIEM detection rules (Rule Management), and ensure that enabled rules are running without errors or warnings (Rule Monitoring).
```

Or maybe go for even more clarity:
```suggestion
.. On the Rules page, re-enable your desired SIEM detection rules (Rule Management tab), and ensure that enabled rules are running without errors or warnings (Rule Monitoring tab).
```

**Review by joepeeples — APPROVED** [review]
Just a few optional suggestions, so went ahead and approved in case you want to merge now. Ping again if you need re-approval!

**Review by joepeeples — APPROVED** [review]

**Review by joepeeples — DISMISSED** [review]

**Comment by 111andre111** [comment]
Looking to the rendered docs page https://security-docs_bk_5830.docs-preview.app.elstc.co/guide/en/security/master/upgrade-intro.html
I have a little general observations question.
We have these headers for upgrades of pre-8.x upgrades:
Screenshot 2024-09-25 at 09 52 03
Screenshot 2024-09-25 at 09 51 47
In comparison to that the 8.x to 8.x Upgrade has a completely other header size level. For me at least from a readability perspective that doesn't sound about right.
In my opinion, there there should be the same separate header level for that section and if there are overlapping sections, these could be linked together, like `third step that needs to be done is x, see section <link to y>`.

Please correct me if I am wrong.

**Comment by natasha-moore-elastic** [comment]
> Looking to the rendered docs page https://security-docs_bk_5830.docs-preview.app.elstc.co/guide/en/security/master/upgrade-intro.html I have a little general observations question. We have these headers for upgrades of pre-8.x upgrades: Screenshot 2024-09-25 at 09 52 03 Screenshot 2024-09-25 at 09 51 47 In comparison to that the 8.x to 8.x Upgrade has a completely other header size level. For me at least from a readability perspective that doesn't sound about right. Screenshot 2024-09-25 at 09 52 17 In my opinion, there there should be the same separate header level for that section and if there are overlapping sections, these could be linked together, like `third step that needs to be done is x, see section <link to y>`.
> 
> Please correct me if I am wrong.

Hi @111andre111, thanks for having a look. **Perform an 8.x to 8.x upgrade on a deployment** is a subsection within the **Upgrade from an earlier 8.x version** section, which is why the heading sizes are different. Both sections talk about the same kind of upgrade (8.x to 8.x)

Essentially, **Upgrade from an earlier 8.x version** consists of the following steps:
1. Plan for your upgrade
2. Pre-upgrade steps
3. Perform the upgrade

Maybe this would be clearer if we renamed **Upgrade from an earlier 8.x version** to **Upgrade from an 8.x to an 8.x version**. This would also be consistent with the other main section titles (**Upgrade from 7.17 to an 8.x version** and **Upgrade from 7.16 or earlier to an 8.x version**).

**Review by joepeeples — APPROVED** [review]

**Pull request merged**